src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -1901,22 +1899,33 @@ protected override void ProcessRecord()
                         byte[] securityDescriptorByte = new byte[rawSecurityDescriptor.BinaryLength];
                         rawSecurityDescriptor.GetBinaryForm(securityDescriptorByte, 0);
 
-                        GCHandle hDacl = GCHandle.Alloc(rawDacl, GCHandleType.Pinned);
-                        lpDacl = hDacl.AddrOfPinnedObject();
-
-                        status = NativeMethods.SetServiceObjectSecurity(
-                                    hService,
-                                    SecurityInfos.DiscretionaryAcl,
-                                    lpDacl);
-
-                        hDacl.Free();
+                        GCHandle hDacl = GCHandle.Alloc(securityDescriptorByte, GCHandleType.Pinned);
+                        try
+                        {
+                            status = NativeMethods.SetServiceObjectSecurity(
+                                        hService,
+                                        SecurityInfos.DiscretionaryAcl,
+                                        hDacl.AddrOfPinnedObject());
 
-                        if (!status)
+                            if (!status)
+                            {
+                                int lastError = Marshal.GetLastWin32Error();
+                                Win32Exception exception = new Win32Exception(lastError);
+                                bool accessDenied = exception.NativeErrorCode == 0x5;
+                                WriteNonTerminatingError(
+                                    service,
+                                    exception,
+                                    nameof(ServiceResources.CouldNotSetServiceSecurityDescriptorSddl),
+                                    StringUtil.Format(ServiceResources.CouldNotSetServiceSecurityDescriptorSddl, Name, exception.Message),
+                                    accessDenied ? ErrorCategory.PermissionDenied : ErrorCategory.WriteError);
```

Comment:
I think default error category should be `ErrorCategory.InvalidOperation` rather than `ErrorCategory.WriteError`.